### PR TITLE
Feature/data flow documentation

### PR DESCRIPTION
**Pull request by mgdaily**
This adds some docs about data flow and how the ingester, ocs_archive and science-archive work together. Also added a section to the science archive component page talking about the ocs_archive and linking to it, and I updated some stale wording.

I think the doc mostly covers what we need, but let me know if there's anything glaring missing!

### CODE REVIEW

**Review by markBowman — APPROVED** [review]
Nicely done

File: components/science_archive.md [hunk 1]
```diff
@@ -6,8 +6,16 @@ subtitle: Django Application that manages Data products that are stored in AWS S
 
 # Science Archive
 
-This Django application manages the observatory's data products, and references the raw data products which are stored in AWS S3. It features a REST API for retrieving metadata and download links for data products, and also for submitting metadata and data products for upload to S3. For the complete list of filterable FITS header values, check out the [API specification]({% link api/downtime.md %}).
+This Django application manages archival and retrieval of an observatory's data products. It features a REST API for retrieving metadata and download links for data products, and also for submitting metadata and data products for upload to a cloud or local filestore. For the complete list of filterable FITS header values, check out the [API specification]({% link api/science_archive.md %}).
```
Comment (jnation3406) [inline components/science_archive.md]: I guess its not quite correct to say the science archive has a REST api for uploading to a cloud or filestore. Instead maybe we should say it has an API for adding images that have been uploaded to a cloud or local filestore...

File: integration/data_flow.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Observatory Data Flow
+
+For users who wish to utilize the [Science Archive](https://github.com/observatorycontrolsystem/science-archive) for data archival and retrieval, this document describes the components necessary.
+
+## Main Components
+
+* [OCS Ingester](https://github.com/observatorycontrolsystem/ingester) - commonly used as a library to simplify the ingestion process of observatory data products into the science archive. 
+* [OCS Science Archive](https://github.com/observatorycontrolsystem/science-archive) - A Django application which aids in the archival storage and retrieval of observatory data products.
+
+Both of these applications make use of a common library, the [OCS Archive Library](https://github.com/observatorycontrolsystem/ocs_archive). The OCS Archive library allows the OCS Ingester and Science Archive to share common configuration and reduces the need for duplicated code across the two projects - this shared code contains classes used to abstract data files and data storage backends. See the "Advanced Topics" for more information about customizing the behavior of the `ocs_archive` library.
```
Comment (jnation3406) [inline integration/data_flow.md]: you could turn the "Advanced Topics" in the last sentence into a link to that section

File: integration/data_flow.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+Because both the OCS Science Archive and OCS Ingester depend on the OCS Archive Library, the two can share common configuration. Runtime configuration for the OCS Archive Library is set using environment variables - see [OCS Archive Configuration Documentation](https://github.com/observatorycontrolsystem/ocs_archive#configuration) for more detail about possible configuration options.
+
+## Telescope Control System Integration
+
+In the execution of an observation, the observatory's Telescope Control System generates data products. For these products to be archived and served to users or staff via an OCS Science Archive, they must be ingested.
+
+### A Simple Ingestion Scheme
+
+Once a TCS has created its final data products to be archived, the TCS may call out to the Ingester library directly. If the TCS is written in Python, the call is as simple as:
```
Comment (jnation3406) [inline integration/data_flow.md]: "the call is as simple as" -> "the call is as simple as setting the relevant environment variables for your filestore type and running" or something like that, just to hammer in that environment variables must be setup for this library to work properly

File: integration/data_flow.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+This mapping can be overridden using the `FILETYPE_MAPPING_OVERRIDES` environment variable if the default is not sufficient or needs to be changed.
+
+
+#### Filestore
+
+A set of classes which represent storage backends exists within the `storage` module of the `ocs_archive` library. 
+
+The base [`FileStore` class](https://github.com/observatorycontrolsystem/ocs_archive/blob/main/ocs_archive/storage/filestore.py#L20) is intended to represent a data product storage backend, and can be sub-classed and extended to work with many different storage solutions.
+
+For example, the [`S3Store`](https://github.com/observatorycontrolsystem/ocs_archive/blob/main/ocs_archive/storage/s3store.py#L23) contains customizations to the base `FileStore` class to allow observatory data products to be stored in [Amazon S3](https://aws.amazon.com/s3/). If an observatory needs to use a different cloud provider, then a new subclass can be created.
```
Comment (jnation3406) [inline integration/data_flow.md]: should we mention the filesystemstore here as well - stores files locally on the file system in a directory structure, and requires an http server serving up the data for users to access it - also stress this isn't as secure so S3 storage should be favored

File: integration/data_flow.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+{"filepath": "/path/to/file/on/disk"}
+```
+
+### Retry Scheme for Data Product Ingestion
+
+For added robustness, it is good practice to integrate retry logic into the data product ingestion process. It's for this reason that the OCS Ingester defines a set of Python [exception classes](https://github.com/observatorycontrolsystem/ingester/blob/main/ocs_ingester/exceptions.py) that will be raised in appropriate situations:
+
+* **RetryError** - Exception that is raised when an error happens that can be retried.
+* **BackoffRetryError** - Exception that is raised when an error happens that can be retried with an expontential backoff. For example, networking latency errors that may succeed at a later time.
+* **NonFatalDoNotRetryError** - Exception that is raised when an error happens that should not be retried and is also not a fatal condition.
```
Comment (jnation3406) [inline integration/data_flow.md]: maybe give an example of this to see how its different from the DoNotRetryError. I think is if the file already exists in the archive.

**Review by jnation3406 — APPROVED** [review]
Looks good!

**Pull request merged**